### ISSUE TITLE
Refine and test setup instructions for Windows

### ISSUE THREAD

**Issue by JohnGuilding** [jzaki/bls-wallet#501]
**As a** Windows user,
**I want to** have clear instructions on how to setup BLS Wallet with WSL2 on a windows machine
**So that** I can run BLS Wallet components with WSL2 easily

Following #500, we should test setting up the BLS Wallet repo on a windows machine using WSL2. After doing this successfully, we should amend the instructions if needed with any additional detail or gotchas.

**Note if you're developing on a linux/mac. You could test this by using a Windows VM and downloading WSL2**

### Acceptance criteria
- Can follow the [local development](https://github.com/web3well/bls-wallet/blob/main/docs/local_development.md) instructions to completion on a Windows machine using WSL2. Completion means running a node, a local aggregator and sending a transaction successfully with Quill.
- Can follow the [remote development](https://github.com/web3well/bls-wallet/blob/main/docs/remote_development.md) instructions to completion on a Windows machine using WSL2. Completion means running a local aggregator pointing at testnet (arbitrum goerli) and sending a transaction successfully with Quill on testnet (arbitrum goerli)
- Setup instructions for [local development](https://github.com/web3well/bls-wallet/blob/main/docs/local_development.md) and [remote development](https://github.com/web3well/bls-wallet/blob/main/docs/remote_development.md) are updated with any additional detail or gotchas.

**Comment by ghost** [comment]
Hi @JohnGuilding can you please elaborate what needs to be done on this issue?

**Comment by JohnGuilding** [comment]
Hey @vishalroy2198! Just added more detail. Would you be interested in picking this up?

Technically, we were going to do #500 first, but as that is so small, and you've responded so quick, you could just go ahead and pick this up and add the additional instructions as part of this :)

**Comment by ri-dev-git** [comment]
Hi @JohnGuilding! 
I was abl to setup the repo in my local development this time! 
I have used wsl 2 with ubuntu distro and have setup the docker settings:
![image](https://user-images.githubusercontent.com/67939540/230588977-3cd2f80c-aa68-4892-858d-62447c4a2c6c.png)

Got the extension running:
![image](https://user-images.githubusercontent.com/67939540/230588686-96f00d62-0e96-4a44-81cb-9d1fd5f09c6f.png)

But when I want to run the integration-tests(with localnode and aggregatot/program.ts) it pops with an error:
![image](https://user-images.githubusercontent.com/67939540/230592116-88d90b64-82c5-4c47-a291-eec224054818.png)

If i am not mistaken it is optimism network with its chain id. Can you help me navigate through it or tell where to make the changes?
Also i want to know how can i make transaction with quill, As there is only 0 eth in the wallet.

Thanks in advance!:)

**Comment by JohnGuilding** [comment]
Hey @ri-dev-git awesome, great to hear that you've got that working!

ok so at the moment, the integration tests run on a geth network. So you'll need to run `yarn start` in the contracts directory first. Then you can run the aggregator locally, you can copy the values from `.env.local.example` into your `.env` file.

Quill on the other hand, we haven't updated that to use the geth node, so in order to use that, you'll need to run `yarn start-hardhat`. Quill also doesn't pay the aggregator fees at the moment. We have both of these issues tracked in https://github.com/web3well/bls-wallet/issues/551 to look at in the near future. So ensure you also set `REQUIRE_FEES` to false instead of true in the aggregator .env when running against quill. This will mean the aggregator will not charge fees to incoming transactions, and quill will work.

> Also i want to know how can i make transaction with quill, As there is only 0 eth in the wallet.

To do this, you'll need to send some eth from your metamask account to Quill on the local network. You'll need to connect metamask to localhost:8545 to do this and import a hardhat account. Let me know if you need any help with that.

So to summarise:

**Integration tests**
- start node and deploy contracts with `yarn start`

**Quill**
- start node and deploy contracts with `yarn start-hardhat`
- Set `REQUIRE_FEES` to false in aggregator `.env`
- Send ETH from metamask to Quill on local network

Take a look at the scripts being run to get an idea of what's going on. Also might be worth re-reading the [local setup docs](https://github.com/web3well/bls-wallet/blob/main/docs/local_development.md#chain-rpc-node) to ensure you're up-to-date with any recent changes we've made.

**Comment by ri-dev-git** [comment]
hey @JohnGuilding! Thanks for the reply. I tried the steps and followed the local env docs.
the transactions are being logged in console:
![image](https://user-images.githubusercontent.com/67939540/231207819-c18643f2-4324-4222-86fc-34a177caf70c.png)

are the eth transferred suppose to update the balance in the wallet? if yes, i am not able to see them :( 
![image](https://user-images.githubusercontent.com/67939540/231213240-50126081-9e14-49f5-8021-38ab5cfb982b.png)

**Comment by JohnGuilding** [comment]
@ri-dev-git trying visiting the settings page in quill and check you're running against a local network, it defaults to arbitrum-goerli. Let me know if that doesn't work

**Comment by ri-dev-git** [comment]
![image](https://user-images.githubusercontent.com/67939540/231735327-1d997e62-1e0d-4d2b-942c-d6b0fa608abf.png)
yes sir! @JohnGuilding  i was able to do it!

I am trying to test setup for remote! Do i need to update the mnemonic with my wallet mnemonic to deploy it to arbitrum?
i am trying to follow the steps in the docs it prompts error as `insufficient funds for intrinsic transaction cost`.
Thank you!

**Comment by JohnGuilding** [comment]
Sweeet

ok so testnet:

1. Switch to the `Arbitrum Goerli` network from the settings page in Quill
2. Try with the default config. If it doesn't work, change your local config (`config.json`) to point to a local aggregator instead
```json
"arbitrum-goerli": {
      ...
      "aggregatorUrl": "http://localhost:3000",
      ...
    },
```

3. Run a local aggregator, but use the .env settings from `.env.example`, instead of `.env.local.example`.
4. Replace `PRIVATE_KEY_AGG` with a private key for a funded account on arbitrum goerli.
5. Set REQUIRE_FEES to false
6. Run the local aggregator with the new settings which point it to arbitrum goerli, and then run Quill.

Let me know how you get on :)

If all else fails, try updating `MAX_GAS_PER_BUNDLE` to a much higher value such as 8000000. Some testnet transactions have been failing recently because the max gas per bundle is going over the default value for this variable. This is something separate that we're looking into 

> Do i need to update the mnemonic with my wallet mnemonic to deploy it to arbitrum?

You shouldn't need to do anything like that, no. The deployment cost will be paid for by the original transaction. Because Quill doesn't yet support paying fees (open issue we're looking at soon), the aggregator will pay the fees for wallet creation when `REQUIRE_FEES` is set to false

**Comment by ri-dev-git** [comment]
![image](https://user-images.githubusercontent.com/67939540/232688832-0e05a062-8613-46da-963b-84c7cb87790c.png)

![image](https://user-images.githubusercontent.com/67939540/232689295-8e0d645b-4182-447f-a694-85c2bbd7e9a5.png)


Hi @JohnGuilding! it all went well i was able to send and receive transactions using the remote development with the config file as is! You can go ahead and close the issue and also can you please assign it to me?

**Comment by JohnGuilding** [comment]
Hey @ri-dev-git just assigned you to the issue. 

> Setup instructions for [local development](https://github.com/web3well/bls-wallet/blob/main/docs/local_development.md) and [remote development](https://github.com/web3well/bls-wallet/blob/main/docs/remote_development.md) are updated with any additional detail or gotchas.

^ Once this acceptance criteria is done, I'll be able to close the issue. If you raise a PR with any updating instructions/tips you gathered along the way, I can review it :)

**Comment by ri-dev-git** [comment]
Yes will sure do!

**Comment by ri-dev-git** [comment]
Local development setup instructions:
- setup works good!
- info about changing the network to localhost in the settings page and transferring the eth from metamask to Quill could help
to have a basic grasp over wallet.

Remote development:
- sometimes while deploying the scripts the rpc prompts with an error of `httttp:provider Error`, it might be due  to the rpc latency and traffic. Its only sometimes.
- Otherwise, the documentation is quite straight forward for remote development!

**Comment by ri-dev-git** [comment]
Hi @JohnGuilding! Here are setup instructions:
- Local Development:
  - Following the local development docs are quite straight forward.
  - Addition of Changing .env file according to env could be helpful for new contributors!
 
- Remote development:
  - I found no problem in following the steps.
  - Addition of Changing .env files according to env could be helpful for new contributors!

**Comment by JohnGuilding** [comment]
Hey @ri-dev-git, awesome thanks.

Would you be able to raise a pull request for this and add it to the docs yourself? I can raise a PR for this myself, but don't want to steal your thunder if you want credit on github for the change.

You could create a branch called `501-refine-and-test-setup-instructions-for-windows`, then raise a PR into main.

It would also be good to know about any issues you had setting up WSL2 to get bls-wallet to work. E.g. did you have any useful learnings from #435?

**Comment by ri-dev-git** [comment]
> It would also be good to know about any issues you had setting up WSL2 to get bls-wallet to work. E.g. did you have any useful learnings from #435?

None as such @JohnGuilding ! Setting up the setting in Docker settings as metionned in https://github.com/web3well/bls-wallet/issues/501#issuecomment-1500156560 and opening the vscode with wsl terminal with a Linux distro was all i needed to do get things running this time!

**Comment by JohnGuilding** [comment]
Ok sweet, that's good to know it was straightforward in the end.

Let me know if you have any issues raising the PR :)
